### PR TITLE
Integration of 'aosp-xenvm-trout' AOSP device

### PR DESCRIPTION
**Pull request by svlad-90**
Integration of 'aosp-xenvm-trout' AOSP device to the open-source

### CODE REVIEW

File: shared/config/init.vendor.rc [hunk 1]
```diff
@@ -0,0 +1,152 @@
+on early-init
```
Comment (arminn) [inline shared/config/init.vendor.rc:1]: Comment to patch "Override the 'init.vendor.rc' file" : Please add into description why we need to override original init.vendor.rc

File: aosp_xenvm_trout_arm64.mk [hunk 2]
```diff
@@ -2,6 +2,13 @@ $(call inherit-product, device/google/trout/aosp_trout_arm64.mk)
 
 include device/epam/aosp-xenvm-trout/aosp_xenvm_trout_common.mk
 
+DEVICE_MANIFEST_FILE := device/google/trout/manifest_x86.xml
```
Comment (arminn) [inline aosp_xenvm_trout_arm64.mk]: Comment to patch "Override trout VHAL implementation" : The patch "Override trout VHAL implementation" should be reworked and divided  into at least 3  parts since there are several not connected changes. Also additional information regarding why audio.primary.caremu overrided should be added

File: aosp_xenvm_trout_common.mk [hunk 3]
```diff
@@ -7,5 +7,10 @@ PRODUCT_COPY_FILES += \
 
 LOCAL_VHAL_PRODUCT_PACKAGE ?= android.hardware.automotive.vehicle@2.0-virtualization-service-extended
 
-PRODUCT_PACKAGES += audio.primary.caremu.extended \
+LOCAL_AUDIO_PROPERTIES ?= \
```
Comment (arminn) [inline aosp_xenvm_trout_common.mk:8]: Comment on patch "Override implementation of emulator's audio driver" : I believe this patch should be a solid part with the audio code from the "Override trout VHAL implementation"

File: sepolicy/genfs_contexts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# x86
+genfscon sysfs /devices/pci0000:00/0000:00:04.0 u:object_r:sysfs_gpu_trout:s0
+
+# arm64
+genfscon sysfs /devices/platform/4010000000.pcie/pci0000:00/0000:00:03.0 u:object_r:sysfs_gpu_trout:s0
```
Comment (arminn) [inline sepolicy/genfs_contexts]: Comment on patch "Extended sepolicy rules" : It looks like patch sets the correct context for virtio gpu and and provides additional configuration for access. Please update description with respect to it.

File: sepolicy/genfs_contexts [hunk 5]
```diff
@@ -2,4 +2,4 @@
 genfscon sysfs /devices/pci0000:00/0000:00:04.0 u:object_r:sysfs_gpu_trout:s0
 
 # arm64
-genfscon sysfs /devices/platform/4010000000.pcie/pci0000:00/0000:00:03.0 u:object_r:sysfs_gpu_trout:s0
+genfscon sysfs /devices/platform/10000000.pcie/pci0000:00/0000:00:03.0 u:object_r:sysfs_gpu_trout:s0
```
Comment (arminn) [inline sepolicy/genfs_contexts:5]: Comment on patch "Adjusted path to PCI device in the genfs_contexts" : this should be integrated into "Extended sepolicy rules" including description

File: aosp_xenvm_trout_common.mk [hunk 6]
```diff
@@ -5,5 +5,7 @@ PRODUCT_COPY_FILES += \
     device/epam/aosp-xenvm-trout/shared/config/init.recovery.rc:$(TARGET_COPY_OUT_RECOVERY)/root/init.recovery.xenvm_trout.rc \
     device/epam/aosp-xenvm-trout/shared/config/ueventd.rc:$(TARGET_COPY_OUT_RECOVERY)/root/ueventd.xenvm_trout.rc
 
+LOCAL_VHAL_PRODUCT_PACKAGE ?= android.hardware.automotive.vehicle@2.0-virtualization-service-extended
```
Comment (arminn) [inline aosp_xenvm_trout_common.mk]: Comment to patch "Override cuttlefish VHAL implementation" : Please provide information in description why this override needed

File: sepolicy/file_contexts [hunk 7]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (arminn) [inline sepolicy/file_contexts]: Comment on patch "Override implementation of emulator's audio driver" : this change not needed.

File: aosp_xenvm_trout_arm64.mk [hunk 8]
```diff
@@ -45,7 +45,7 @@ include device/epam/aosp-xenvm-trout/aosp_xenvm_trout_common.mk
 DEVICE_MANIFEST_FILE := device/google/trout/manifest_x86.xml
 DEVICE_MATRIX_FILE += device/google/trout/compatibility_matrix.xml
 DEVICE_FRAMEWORK_COMPATIBILITY_MATRIX_FILE := device/google/trout/framework_compatibility_matrix.xml
-
```
Comment (arminn) [inline aosp_xenvm_trout_arm64.mk]: Comment on patch "Extend rootfs size to 2Gb" : this change not needed.

**Comment by arminn** [comment]
The patches "Override cuttlefish VHAL implementation" "Deploy vsock/grpc backends" "Revert deletion of overriden_impl" : adding some stuff, removing it and adding again. This should be reworked.

**Comment by arminn** [comment]
Note: I did per commit review, for all code comments there is a comment which patch was addressed(github very strangely displays all of that).

File: shared/config/init.vendor.rc (hunk 1 above)
Comment (svlad-90) [reply shared/config/init.vendor.rc:1]: Hi @arminn,

I've checked why this change was done. It was introduced in order to prevent the renaming of the eth0 interface during the startup of the system. I've extended the commit message.

Done.

File: aosp_xenvm_trout_arm64.mk (hunk 2 above)
Comment (svlad-90) [reply aosp_xenvm_trout_arm64.mk]: The change was split into 3 commits:

https://github.com/xen-troops/android_device_epam_xenvm-trout/pull/1/commits/397fd4d1f4b8672a5aeac3dd015afe0429d032bb

https://github.com/xen-troops/android_device_epam_xenvm-trout/pull/1/commits/647156544e7a19d2a8949886d6c0fec5a23b84c8

https://github.com/xen-troops/android_device_epam_xenvm-trout/pull/1/commits/50e2fab5be43eb7e5f978a10a81e12b08ccb3b7f

The sum of all 3 changes is equal to the previous bif commit, except that I've removed the following 3 lines:

```
DEVICE_MANIFEST_FILE := device/google/trout/manifest_x86.xml
DEVICE_MATRIX_FILE += device/google/trout/compatibility_matrix.xml
DEVICE_FRAMEWORK_COMPATIBILITY_MATRIX_FILE := device/google/trout/framework_compatibility_matrix.xml
```

I don't remember why they were needed and was not able to explain the motivation. It seems that they are not needed for arm64. After all changes for all your comments will be done, I'll do a testing session. If these 3 lines would have a critical meaning for having the system up and running. I'll return them back.

Done.

File: aosp_xenvm_trout_common.mk (hunk 6 above)
Comment (svlad-90) [reply aosp_xenvm_trout_common.mk]: Hi @arminn,

New commit: https://github.com/xen-troops/android_device_epam_xenvm-trout/pull/1/commits/a38d2538259ba6467c7f6dfcd0fed6365d65316b

I've extended its description.

In short - all this is done to comment these 2 lines:

```
initObd2LiveFrame(*mPropStore->getConfigOrDie(OBD2_LIVE_FRAME));
initObd2FreezeFrame(*mPropStore->getConfigOrDie(OBD2_FREEZE_FRAME));
```

A more detailed explanation is in the commit message.

Done.

File: aosp_xenvm_trout_common.mk (hunk 3 above)
Comment (svlad-90) [reply aosp_xenvm_trout_common.mk:8]: Hi @arminn, this patch was moved to the following commits:

https://github.com/xen-troops/android_device_epam_xenvm-trout/pull/1/commits/50e2fab5be43eb7e5f978a10a81e12b08ccb3b7f

https://github.com/xen-troops/android_device_epam_xenvm-trout/pull/1/commits/b94032ae0a0317d205ede8e4b34fbcf112a77177 ( there was one last line in this commit related to vehicle topics )

This patch was removed, as it is empty now.

Done.

File: sepolicy/file_contexts (hunk 7 above)
Comment (svlad-90) [reply sepolicy/file_contexts]: Hi @arminn,

Done.

File: aosp_xenvm_trout_arm64.mk (hunk 8 above)
Comment (svlad-90) [reply aosp_xenvm_trout_arm64.mk]: Hi @arminn, it is hard to follow this comment after some work is done ( GitHub screws it up ). But it seems to be removed now.

Done.

File: sepolicy/genfs_contexts (hunk 4 above)
Comment (svlad-90) [reply sepolicy/genfs_contexts]: Hi @arminn,

Done.

File: sepolicy/genfs_contexts (hunk 5 above)
Comment (svlad-90) [reply sepolicy/genfs_contexts:5]: Hi @arminn, the previous value was compatible with QEMU. While the new one is compatible with the physical device.

I think we want to have both those values in our commit history in order to be able to "remember" them, Someone might want to use QEMU. Then he might fetch the previous value form the history.

Let me know, whether you agree with this.

**Comment by svlad-90** [comment]
Hi @arminn, I've lost one of your comments in GitHub. The one, related to squashing all those "remove, and then add again the overriden_impl". So firing another stand-alone comment - it is done. The redundant patch was removed.

**Comment by svlad-90** [comment]
Hi @arminn, seems that all items are done. I'll spend tomorrow in order to check whether the final result is still a working piece of SW. ))

**Comment by svlad-90** [comment]
Hi @arminn, it was checked that the Android start-up is OK after applying all the refactored source code.

File: sepolicy/genfs_contexts (hunk 5 above)
Comment (arminn) [reply sepolicy/genfs_contexts:5]: Ok make sense

**Comment by arminn** [comment]
@svlad-90 add plz after your signoff to patches:

Reviewed-by: Andrii Chepurnyi \<andrii_chepurnyi@epam.com\>

and I will merge PR

**Comment by svlad-90** [comment]
Hi @arminn, I've added the reviewed-by tag to all commits. But, please, do not merge it yet. I want to add a fix for the issue with multiple displays to this PR. Once it's ready - I'll let you know with a separate comment.

**Comment by svlad-90** [comment]
Hi @arminn, ready for merge.
